### ISSUE TITLE
[BUG] include env_file not acting as documented

### ISSUE THREAD

**Issue by designermonkey** [docker/compose#11002]
### Description

From the manual:
> `env_file` defines an environment file(s) to use to define default values when interpolating variables in the Compose file being parsed. It defaults to `.env` file in the `project_directory` for the Compose file being parsed.

The way this reads is that if I pass an env file, those values will be interpolated into the included docker-compose file, not that they will be the values passed to the run command environment. This is because of, specifically **use to define default values when interpolating variables in the Compose file**.

This is not what is actually happening.

Docker compose loads the `.env` file local to the called `docker-compose.yml` file as normal, but doesn't load the `.env` file local to the included `docker-compose.yml` file as described above.

### Steps To Reproduce

Please see https://github.com/designermonkey/dc-bug-example

The variable `SWOOLE_PORT` is expected to be interpolated as described in the docs but is not.

```
$ docker compose build                                      
WARN[0000] The "SWOOLE_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "SWOOLE_PORT" variable is not set. Defaulting to a blank string. 
1 error(s) decoding:

* error decoding 'ports': No port specified: :<empty>
```

### Compose Version

```Text
Compose: v2.20.2-desktop.1
```

### Docker Environment

```Text
Client:
 Version:    24.0.5
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     /Users/porterj/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     /Users/porterj/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/porterj/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/porterj/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     /Users/porterj/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/porterj/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/porterj/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     /Users/porterj/.docker/cli-plugins/docker-scout

Server:
 Containers: 38
  Running: 34
  Paused: 0
  Stopped: 4
 Images: 22
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: 9d0b95ee-c75a-40eb-b5ea-61a46903352f
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Anything else?

This is the one feature I wanted to help me away from using `Makefile` hacks to pre-load the environment variables. I hope it is considered a valid bug that will be fixed and not that the documentation gets re-written to remove the interpolation reference, as that would be highly saddening.

I hope I provided enough info, if any more info is required, I will gladly share it.

**Comment by ndeloof** [comment]
`env_file` ATTRIBUTE in the compose file is used to set container environment, this is the same as `docker run --env-file`
`--env-file` FLAG is used to select file used for variable interpolation (defaults to `.env`)

Don't worry, this is a pretty common confusion

**Comment by ndeloof** [comment]
> From the manual:

Where did you read this quote ?

**Comment by designermonkey** [comment]
I know it applies to `include`; that's why it's in the bug title. I probably wasn't clear, but my example repo shows exactly what I mean.

This quote specifically mentions interpolation in the compose file, not the running containers. Is there no way it can be applied to the compose file as well? How are we supposed to pass variables into the file for it to act like we were loading it locally to our `.env` file?

**Comment by ndeloof** [comment]
ok sorry, didn't read carefully enough

**Comment by ndeloof** [comment]
duplicate for https://github.com/docker/compose/issues/10945, fixed in  v2.20.3

**Comment by designermonkey** [comment]
Oh brilliant, thanks @ndeloof :)
